/**
 * Messages for the DDSketch module
 */


/**
 * The Module DDSketch
 */
 message DDSketchArg {
  double accuracy = 1;
  uint64 max_bucket_number = 2;
  uint64 offset = 3;
}


/**
* Clears the captured data.
*/
message DDSketchCommandEmptyArg{}


/**
* Returns the status of the collected data.
*/
message DDSketchCommandGetStatArg{}


/**
* Contains the status of the collected data.
*/
message DDSketchCommandGetStatResponse{
  uint64 packet_number = 1;
  uint64 bucket_number = 2;
  uint64 max_bucket_number = 3;
  double accuracy = 4;
  double lambda = 5;
}

/**
* Returns the status of the collected data.
*/
message DDSketchCommandGetContentArg{}


/**
* Contains the status of the collected data.
*/
message DDSketchCommandGetContentResponse{
  message Bucket{
    uint64 index = 1;
    uint64 counter = 2;
  }
  repeated Bucket content = 1;
}
